File: .codex/skills/project-scaffolding/assets/nestjs-prisma-postgres/src/auth/auth.controller.ts
import { Body, Controller, Get, Post, Req, UseGuards } from "@nestjs/common";
import { AuthService } from "./auth.service";
import { AuthGuard } from "./auth.guard";

@Controller("auth")
export class AuthController {
  constructor(private readonly authService: AuthService) {}

  @Post("register")
  register(
    @Body() body: { email: string; password: string; name?: string }
  ) {
    return this.authService.register(body);
  }

  @Post("login")
  login(@Body() body: { email: string; password: string }) {
    return this.authService.login(body);
  }

  @Post("refresh")
  refresh(@Body() body: { refreshToken: string }) {
    return this.authService.refresh(body.refreshToken);
  }

  @Post("logout")
  async logout(@Body() body: { refreshToken: string }) {
    if (body.refreshToken) {
      await this.authService.revokeRefreshToken(body.refreshToken);
    }
    return { status: "ok" };
  }

  @UseGuards(AuthGuard)
  @Get("me")
  me(@Req() req: { user: { sub: string } }) {
    return { user: req.user };
  }
}

File: .codex/skills/project-scaffolding/assets/java-spring-postgres/src/main/java/com/codexkit/template/controller/AuthController.java
package com.codexkit.template.controller;

import com.codexkit.template.domain.User;
import com.codexkit.template.service.AuthService;
import java.util.Map;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/auth")
public class AuthController {
  private final AuthService authService;

  public AuthController(AuthService authService) {
    this.authService = authService;
  }

  @PostMapping("/register")
  public Map<String, Object> register(@RequestBody Map<String, String> payload) {
    return authService.register(payload.get("email"), payload.get("password"), payload.get("name"));
  }

  @PostMapping("/login")
  public Map<String, Object> login(@RequestBody Map<String, String> payload) {
    return authService.login(payload.get("email"), payload.get("password"));
  }

  @PostMapping("/refresh")
  public Map<String, Object> refresh(@RequestBody Map<String, String> payload) {
    return authService.refresh(payload.get("refresh_token"));
  }

  @GetMapping("/me")
  public Map<String, Object> me(@RequestHeader("Authorization") String authorization) {
    User user = authService.requireUser(authorization);
    return Map.of("user", Map.of("id", user.getId(), "email", user.getEmail(), "role", user.getRole()));
  }
}

